src/pages/workspaces/workspace/Dashboard/Dashboard.ts

```diff
@@ -0,0 +1,662 @@
+import { TooltipTrigger } from '@terra-ui-packages/components';
+import { cond } from '@terra-ui-packages/core-utils';
+import _ from 'lodash/fp';
+import {
+  CSSProperties,
+  ForwardedRef,
+  Fragment,
+  PropsWithChildren,
+  ReactNode,
+  useCallback,
+  useEffect,
+  useImperativeHandle,
+  useMemo,
+  useState,
+} from 'react';
+import { div, h, h3, i, span } from 'react-hyperscript-helpers';
+import * as breadcrumbs from 'src/components/breadcrumbs';
+import Collapse from 'src/components/Collapse';
+import { ButtonPrimary, ButtonSecondary, LabeledCheckbox, Link, spinnerOverlay } from 'src/components/common';
+import { centeredSpinner, icon, spinner } from 'src/components/icons';
+import { InfoBox } from 'src/components/InfoBox';
+import { MarkdownEditor, MarkdownViewer } from 'src/components/markdown';
+import { SimpleTable } from 'src/components/table';
+import { WorkspaceTagSelect } from 'src/components/workspace-utils';
+import { Ajax } from 'src/libs/ajax';
+import { refreshTerraProfile } from 'src/libs/auth';
+import { getEnabledBrand } from 'src/libs/brand-utils';
+import colors from 'src/libs/colors';
+import { reportError, withErrorReporting } from 'src/libs/error';
+import Events from 'src/libs/events';
+import * as Nav from 'src/libs/nav';
+import { getLocalPref, setLocalPref } from 'src/libs/prefs';
+import { forwardRefWithName, useCancellation, useOnMount, useStore } from 'src/libs/react-utils';
+import { authStore } from 'src/libs/state';
+import * as Style from 'src/libs/style';
+import { append, formatBytes, newTabLinkProps, withBusyState } from 'src/libs/utils';
+import {
+  canEditWorkspace,
+  canWrite,
+  isGoogleWorkspace,
+  isOwner,
+  WorkspaceAccessLevel,
+  WorkspaceACL,
+} from 'src/libs/workspace-utils';
+import SignIn from 'src/pages/SignIn';
+import { CloudInformation } from 'src/pages/workspaces/workspace/Dashboard/CloudInformation';
+import { WorkspaceInformation } from 'src/pages/workspaces/workspace/Dashboard/WorkspaceInformation';
+import DashboardPublic from 'src/pages/workspaces/workspace/DashboardPublic';
+import { displayConsentCodes, displayLibraryAttributes } from 'src/pages/workspaces/workspace/library-attributes';
+import { InitializedWorkspaceWrapper as Workspace, StorageDetails } from 'src/pages/workspaces/workspace/useWorkspace';
+import { wrapWorkspace } from 'src/pages/workspaces/workspace/WorkspaceContainer';
+
+const styles: Record<string, CSSProperties> = {
+  authDomain: {
+    padding: '0.5rem 0.25rem',
+    marginBottom: '0.25rem',
+    backgroundColor: colors.dark(0.15),
+    ...Style.noWrapEllipsis,
+  },
+  tag: {
+    padding: '0.25rem',
+    margin: '0.15rem',
+    backgroundColor: colors.dark(0.15),
+    borderRadius: 10,
+    overflow: 'hidden',
+    wordWrap: 'break-word',
+  },
+};
+
+const displayAttributeValue = (v: unknown): string => {
+  return cond(
+    [_.isArray(v), () => (v as string[]).join(', ')],
```

Comment:
Not sure if something like this would allow avoiding the cast.

```suggestion
    [_.isArray(v), () => v.map((item) => `${item}`).join(', ')],
```

Reply:
It doesn't - typescript still doesn't recognize the type as being an array or even an object.

Reply:
Ah, I was focusing on the `string` part, not thinking about TS not recognizing that it's an array at all.